server/bundler/plugins-loader.js

```diff
@@ -0,0 +1,61 @@
+const existsSync = require( 'fs' ).existsSync;
+const readdirSync = require( 'fs' ).readdirSync;
+const path = require( 'path' );
+const camelCase = require( 'lodash' ).camelCase;
+const upperFirst = require( 'lodash' ).upperFirst;
+
+function getReducers( plugins = {} ) {
+	return plugins
+		.map( ( plugin ) => {
+			if ( existsSync( path.join( 'client', 'plugins', plugin, 'state', 'reducer.js' ) ) ) {
+				return `'${ plugin }': require( 'plugins/${ plugin }/state/reducer' )\n`;
+			}
+		} )
+		.filter( reducer => !! reducer );
+}
+
+function getDecorators( plugins = {} ) {
+	return plugins.map( ( plugin ) => {
+		const decoratorsPath = path.join( 'client', 'plugins', plugin, 'decorators' );
+		if ( existsSync( decoratorsPath ) ) {
+			const decorators = readdirSync( decoratorsPath );
+			const decoratorObjects = decorators.map( decorator => {
+				const decoratorName = upperFirst( camelCase( decorator.split( '.' )[ 0 ] ) );
+				return `\t'${ decoratorName }': require( 'plugins/${ plugin }/decorators/${ decorator }' )\n`;
+			} );
+
+			return `'${ plugin }': {
+				${ decoratorObjects }
+				}`;
+		}
+	} );
+}
```

Comment:
Noticing that this function doesn't return a true array of strings, as it may contain `undefined` entries (when the `existsSync` condition fails). This is relevant because my own tests with ES6 string interpolation show that

```js
`return {
  ${ [ 'pluginA: {}', 'pluginB: {}', undefined, 'pluginC: {}' ] }
};`
```

will evaluate to

```js
return {
  pluginA: {},pluginB: {},,pluginC: {}
};
```

that second comma before `pluginC` being an unexpected token.

Filtering for truthiness is enough to fix this.